dbms/src/Storages/DeltaMerge/Remote/Serializer.cpp

```diff
@@ -179,37 +182,46 @@ SegmentSnapshotPtr Serializer::deserializeSegment(
 
 RepeatedPtrField<RemotePb::ColumnFileRemote> Serializer::serializeColumnFileSet(
     const ColumnFileSetSnapshotPtr & snap,
-    MemTrackerWrapper & mem_tracker_wrapper)
+    MemTrackerWrapper & mem_tracker_wrapper,
+    bool need_mem_data)
 {
     RepeatedPtrField<RemotePb::ColumnFileRemote> ret;
     ret.Reserve(snap->column_files.size());
     for (const auto & file : snap->column_files)
     {
-        if (auto * cf_in_mem = file->tryToInMemoryFile(); cf_in_mem)
-        {
-            ret.Add(serializeCFInMemory(*cf_in_mem));
-        }
-        else if (auto * cf_tiny = file->tryToTinyFile(); cf_tiny)
-        {
-            ret.Add(serializeCFTiny(*cf_tiny, snap->getDataProvider()));
-        }
-        else if (auto * cf_delete_range = file->tryToDeleteRange(); cf_delete_range)
-        {
-            ret.Add(serializeCFDeleteRange(*cf_delete_range));
-        }
-        else if (auto * cf_big = file->tryToBigFile(); cf_big)
-        {
-            ret.Add(serializeCFBig(*cf_big));
-        }
-        else
-        {
-            RUNTIME_CHECK_MSG(false, "Unknown ColumnFile, type={}", magic_enum::enum_name(file->getType()));
-        }
+        ret.Add(serializeCF(file, snap->getDataProvider(), need_mem_data));
         mem_tracker_wrapper.alloc(ret.rbegin()->SpaceUsedLong());
     }
     return ret;
 }
 
+RemotePb::ColumnFileRemote Serializer::serializeCF(
+    const ColumnFilePtr & cf,
+    const IColumnFileDataProviderPtr & data_provider,
+    bool need_mem_data)
+{
+    if (auto * cf_in_mem = cf->tryToInMemoryFile(); cf_in_mem)
+    {
+        return serializeCFInMemory(*cf_in_mem, need_mem_data);
+    }
+    else if (auto * cf_tiny = cf->tryToTinyFile(); cf_tiny)
+    {
+        return serializeCFTiny(*cf_tiny, data_provider);
+    }
+    else if (auto * cf_delete_range = cf->tryToDeleteRange(); cf_delete_range)
+    {
+        return serializeCFDeleteRange(*cf_delete_range);
+    }
+    else if (auto * cf_big = cf->tryToBigFile(); cf_big)
+    {
+        return serializeCFBig(*cf_big);
+    }
+    else
+    {
+        RUNTIME_CHECK_MSG(false, "Unknown ColumnFile, type={}", static_cast<UInt32>(cf->getType()));
+    }
```

Comment:
Just check it without if?
```suggestion
        RUNTIME_CHECK_MSG(false, "Unknown ColumnFile, type={}", static_cast<UInt32>(cf->getType()));
```